tokens/batched/batched_test.go

```diff
@@ -0,0 +1,1015 @@
+package batched
```

Comment:
This test file could be shorter using a combination of generics and interfaces.
Let's leave it as is for now, let's open an issue to refactor this file later.

Reply:
I agree. For posterity's sake, I tried using generics, and the issue with Go generic is types have to be known at compile time, which is exactly what we don't know (given type is a function of test vectors). There were no real gain. Interfaces seems to be more onpoint for readability, but either require a refactor of existing types (which I don't want to do here), or long boiler plate in the test (which I don't like either).

I'm ok to open an issue and actually refactor other code pieces, which I think is the best. Refactor could include a general interface with token request as `[]byte`, instead of type specific token request.

Reply:
created #48